src/NuGet.Services.AzureSearch/Wrappers/SingleSearchResultPage.cs

```diff
@@ -14,7 +14,16 @@ public SingleSearchResultPage(IReadOnlyList<SearchResult<T>> values, long? count
             Count = count;
         }
 
+        public SingleSearchResultPage(IReadOnlyList<SearchResult<T>> values, long? count, IDictionary<string, IList<FacetResult>> facets)
+        {
+            Values = values;
+            Count = count;
+            Facets = facets;
+        }
+
         public IReadOnlyList<SearchResult<T>> Values { get; set; }
         public long? Count { get; }
+        public IDictionary<string, IList<FacetResult>> Facets { get; set; }
```

Comment:
Should this use read only collections?

```suggestion
        public IReadOnlyDictionary<string, IReadOnlyList<FacetResult>> Facets { get; set; }
```

Reply:
The Datatype for facets in response from SDK client is **IDictionary<string, IList<FacetResult>> Facets** . I need to manually convert all the lists into the read-only list for each entry in the dictionary while creating the SingleSearchResultPage object in the searchClientWrapper.
<img width="680" alt="ScreenShot_17-08-2022_18_42_26" src="https://user-images.githubusercontent.com/102016390/185145035-38b30180-9167-4887-b5ea-72d786a332d8.png">